teachertool/src/services/fileSystemService.ts

```diff
@@ -0,0 +1,46 @@
+import { logError } from '../services/loggingService';
+import { ErrorCode } from '../types/errorCode';
+import { Rubric } from '../types/rubric';
+
+export function sanitizeFileName(fileName: string): string {
+    // Remove invalid characters
+    let sanitized = fileName.replace(/[<>:"/\\|?*]+/g, '');
+
+    // Replace spaces with underscores
+    sanitized = sanitized.replace(/ /g, '_');
+
+    return sanitized;
+}
+
+// Serializes the active rubric and writes it to a file.
+// Returns true if the file was written successfully, false otherwise.
+export function writeRubricToFile(rubric: Rubric): boolean {
+    if (!rubric.name || rubric.criteria.length === 0) {
+        return false;
+    }
+
+    const fileName = sanitizeFileName(rubric.name) + '.json';
```

Comment:
Can `rubric.name` be empty?

Reply:
Oh, I see the conditional above. Let's not make a name strictly required. Have it assign a default name to the file if empty: "unnamed-rubric.json".

Reply:
Another approach: give the rubric a default name from the beginning, and not let it become null/empty. Either way, I think we should provide a filename if necessary to allow the export to succeed.

Reply:
I'm happy to have it be exported as unnamed-rubric for now if it hasn't been set yet